### PR TITLE
Improved NodeJS stream declarations & implemented tty and process module declarations

### PR DESCRIPTION
**Pull request by TiddoLangerak**
Title says it all. The module definitions of these modules should now be complete.

### CODE REVIEW

**Comment by TiddoLangerak** [comment]
You're probably all busy with more important features for flow, but I'd love to see this merged such that there is better native support for NodeJS. Is there anything I can do to make it easier to review this for you guys?

**Comment by lightsofapollo** [comment]
+1

File: lib/node.js [hunk 1]
```diff
@@ -669,20 +663,24 @@ declare module "https" {
...
+declare class net$Socket extends stream$Duplex {
+  //TODO
+}
+
+declare class net$Server extends events$EventEmitter {
+  listen(port: number, hostname?: string, backlog?: number, callback?: Function): net$Server;
+  listen(path: string, callback?: Function): net$Server;
+  listen(handle: Object, callback?: Function): net$Server;
+  close(callback?: Function): net$Server;
+  address(): { port: number; fmaily: string; address: string; };
```
Comment (mroch) [inline lib/node.js]: `family` typo, will fix while merging

File: lib/node.js [hunk 2]
```diff
@@ -669,20 +663,24 @@ declare module "https" {
...
+}
+
 declare module "net" {
 
-  declare class Server extends events$EventEmitter {
-    listen(port: number, hostname?: string, backlog?: number, callback?: Function): Server;
-    listen(path: string, callback?: Function): Server;
-    listen(handle: Object, callback?: Function): Server;
-    close(callback?: Function): Server;
-    address(): { port: number; fmaily: string; address: string; };
```
Comment (mroch) [inline lib/node.js]: d'oh, it was already that way

**Comment by mroch** [comment]
thanks! this is way better. sorry this took so long to get to :/

**Comment by mroch** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**